### DISCUSSION TITLE
Using webhooks to approve/deny registration

### DISCUSSION THREAD

**Discussion by auctumnus** [ory/kratos#1978] [Q&A] [answered]
Hello! I'm looking at using Kratos as the auth layer for another API, and I want to ensure a valid user can be created in the API's database when a user is created with the self-service flow in Kratos. I saw that using a registration webhook and responding with an HTTP code over 300, I can decide whether to approve or deny the flow - but the user still gets created in Kratos, since subsequently attempting to register under the same email/username will return an error saying that a user with that identifier already exists. I thought the before hook might be able to do this, but the before hook doesn't actually carry any information about the user attempting to register?
In a slightly less dense way: The after-registration webhook will still write the user to the Kratos database, and there's no hook that can prevent this, as far as I can tell. Is there any way to approve or deny registration based on the identity information?

**Comment by aeneasr** [comment, marked as answer]
Hello there, this is currently being worked on: https://github.com/ory/kratos/pull/1585
